File: frontend/vigie-du-mensonge/src/core/models/articleReview.ts
import type {ArticleStatus} from "@/core/models/articleStatus.ts";

export type ArticleReviewJson = {
    moderatorTag?: string;
    notes?: string;
    decision?: ArticleStatus;
}

export class ArticleReview {
    public moderatorTag?: string;
    public notes?: string;
    public decision?: ArticleStatus;

    constructor(moderatorTag?: string, notes?: string, decision?: ArticleStatus) {
        this.moderatorTag = moderatorTag;
        this.notes = notes;
        this.decision = decision;
    }

    static fromJson(json: ArticleReviewJson): ArticleReview {
        return new ArticleReview(json.moderatorTag, json.notes, json.decision);
    }
}

File: backend/core/dto/response_dto/article_review.go
package response_dto

import "vdm/core/models"

type ArticleReview struct {
	ModeratorTag string               `json:"moderatorTag,omitempty"`
	Decision     models.ArticleStatus `json:"decision,omitempty"`
	Notes        string               `json:"notes,omitempty"`
}

func NewArticleReview(entity models.ArticleReview) ArticleReview {
	dto := ArticleReview{
		Decision: entity.Decision,
		Notes:    entity.Notes,
	}

	if entity.Moderator != nil {
		dto.ModeratorTag = entity.Moderator.Tag
	}

	return dto
}
